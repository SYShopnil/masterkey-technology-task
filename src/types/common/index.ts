export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EDataTestId {
  CAlphabetApp = "CAlphabetApp",
  CAlphabetGrid = "CAlphabetGrid",
  CLayout = "CLayout",
  CPartition = "CPartition",
  SHeaderMain = "SHeaderMain",
  SIconWithMessage = "SIconWithMessage",
  SLoading = "SLoading",
}
