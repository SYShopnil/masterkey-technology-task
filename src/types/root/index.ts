export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
