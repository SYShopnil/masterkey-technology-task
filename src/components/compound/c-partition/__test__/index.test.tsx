import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import CLayout from "../../c-partition-layout";
import { CPartition, PartitionProps } from "..";

describe("When Partition Component Rendered", () => {
  it("Expect render successfully", () => {
    const mock_data: PartitionProps = {
      color: "FFFF",
      id: 12,
      onRemove: () => {},
      onSplit: () => {},
      style: { padding: "25px" },
    };
    render(<CPartition {...mock_data} />);
    const myElement = screen.getByRole(EDataTestId.CPartition);
    expect(myElement).toMatchSnapshot();
  });
});
