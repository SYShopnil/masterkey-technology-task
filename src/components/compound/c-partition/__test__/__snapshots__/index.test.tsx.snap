// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Partition Component Rendered Expect render successfully 1`] = `
<div
  class="relative border"
  role="CPartition"
  style="padding: 25px;"
>
  <div
    class="absolute inset-0 flex items-center justify-center space-x-2"
  >
    <button
      class="px-2 py-1 bg-white border rounded"
    >
      V
    </button>
    <button
      class="px-2 py-1 bg-white border rounded"
    >
      H
    </button>
    <button
      class="px-2 py-1 bg-white border rounded"
    >
      X
    </button>
  </div>
  <div
    class="absolute bottom-0 right-0 w-4 h-4 bg-gray-400 cursor-se-resize"
  />
</div>
`;
